AnimatedWordCloud/Animator/AllocationCalculator/AnimatetdAllocationCalculator.py

```diff
@@ -39,53 +81,72 @@ def add_key_in_allocation_frame(
     :param  AllocationInFrame from_allocation_frame, to_allocation_frame: start frame and end frame
     :param list[str] from_words_to_be_added_key, to_words_to_be_added_key: the keys to be added
     :return: AllocationInFrame from_allocation_frame, to_allocation_frame: start and end frame added to the necessary keys
+    :rtype: tuple[AllocationInFrame, AllocationInFrame]
     """
     for to_be_added_key in from_words_to_be_added_key:
         # add key in from_allocation_frame
-        (font_size, (left_top)) = to_allocation_frame[to_be_added_key]
+        (font_size, left_top) = to_allocation_frame[to_be_added_key]
         from_allocation_frame.add(to_be_added_key, font_size, left_top)
 
     for to_be_added_key in to_words_to_be_added_key:
         # add key in to_allocation_frame
-        (font_size, (left_top)) = from_allocation_frame[to_be_added_key]
+        (font_size, left_top) = from_allocation_frame[to_be_added_key]
         to_allocation_frame.add(to_be_added_key, font_size, left_top)
 
     return from_allocation_frame, to_allocation_frame
 
 
-def calc_frame_value(
-    from_value: float, to_value: float, index: int, n_frames: int
-):
-    # calculate interpolation's value
-    # Linear only for now
-    value = from_value + index / (n_frames + 1) * (to_value - from_value)
+def _calc_frame_value(
+    from_value: float, to_value: float, index: int, config: Config
+) -> float:
+    """
+    :param float from_value, to_value: font_size, x_position, or y_position
+    :param int index: interpolation frame index
+    :param int n_frames: the number of all the interpolation frames
+    :return: calculated interpolation's value
+    :rtype: float
+
+    Linear only for now
+    Add non-Linear interpolation processes in the future.
+    """
+    if config.interpolation_method == "linear":
+        value = from_value + index / (config.n_frames + 1) * (
+            to_value - from_value
+        )
```

Comment:
Separate the logic of each `allocation_method` as different method / module.
`_calc_frame_value` will take all methods in one function